### PR TITLE
fix and simplify devtool build_devctr; upgrade python dependencies

### PR DESCRIPTION
**Pull request by alindima**
# Reason for This PR

We want to regularly update python package dependencies.

## Description of Changes

View commit messages

- [ ] This functionality can be added in [`rust-vmm`][1].

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The issue which led to this PR has a clear conclusion.
- [x] This PR follows the solution outlined in the related issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes follow the [Runbook for Firecracker API changes][2].
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

[1]: https://github.com/rust-vmm
[2]: ../docs/api-change-runbook.md

### CODE REVIEW

**Comment by gc-plp** [comment]
I think that you can avoid specifying the `encoding` for the `open()` calls as long as there a locale set.

Doesn't work great on Windows, but with Linux hosts this is fine.

File: tests/framework/utils.py [hunk 1]
```diff
@@ -193,8 +193,8 @@ def with_arg(self, flag, value=""):
     def build(self):
         """Build the command."""
         cmd = self._bin_path + " "
-        for flag in self._args:
-            cmd += flag + " " + "{}".format(self._args[flag]) + " "
+        for (flag, value) in self._args.items():
+            cmd += flag + " " + "{}".format(value) + " "
```
Comment (gc-plp) [inline tests/framework/utils.py:197]: Would look nicer with `f-strings`.

Comment (alindima) [reply tests/framework/utils.py:197]: yes, most of the `.format`'s should be replaced by f-strings, but it's a titanic amount of work to do this :)

**Comment by alindima** [comment]
> I think that you can avoid specifying the `encoding` for the `open()` calls as long as there is a locale set.
> 
> Doesn't work great on Windows, but with Linux hosts this is fine.

It's the pylint complaining..

File: tests/framework/utils.py (hunk 1 above)
Comment (alindima) [reply tests/framework/utils.py]: Changed this specific one, since it indeed looks guite ugly

**Review by AlexandruCihodaru — APPROVED** [review]

**Review by dianpopa — APPROVED** [review]

**Pull request merged**